services/src/main/java/org/exoplatform/addons/gamification/service/dto/configuration/BadgeDTO.java

```diff
@@ -80,6 +83,8 @@ public BadgeDTO(BadgeEntity badgeEntity) {
 
         this.domain = badgeEntity.getDomain();
 
+        this.domainDTO = new DomainDTO(badgeEntity.getDomainEntity());
```

Comment:
Should test on DomainEntity Existance to avoid NPE

Reply:
It's already done in the second commit :)